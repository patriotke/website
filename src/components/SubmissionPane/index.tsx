/**
 *
 * SubmissionPane
 *
 */

'use client';

import SubmissionToolbar from 'components/SubmissionToolbar';
import { getTopics } from 'lib/topicLib';
import { useState } from 'react';
import MissingPersonForm from 'components/MissingPersonForm';

// type ISubmissionPaneProps = {
// onSubmit: () => void;
// };
function SubmissionPane() {
  const topics = getTopics();
  const [topic, setTopic] = useState('');

  const handleChange = (field: string, value: string) => {
    if (field === 'topic') {
      setTopic(value);
    }
  };
  return (
    <div>
      <SubmissionToolbar topics={topics} onChange={handleChange} />
      {(topic === 'missing-person' || true) && <MissingPersonForm />}
    </div>
  );
}

export default SubmissionPane;
