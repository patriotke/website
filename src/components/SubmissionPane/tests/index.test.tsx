/* eslint-disable import/no-extraneous-dependencies */
import '@testing-library/jest-dom';
// eslint-disable-next-line @typescript-eslint/no-unused-vars
import { render, screen } from '@testing-library/react';
import SubmissionPane from '../index';

describe('SubmissionPane', () => {
  it('renders SubmissionPane', () => {
    render(<SubmissionPane />);

    const { container } = render(<SubmissionPane />);
    expect(container).toMatchSnapshot();
  });
});
