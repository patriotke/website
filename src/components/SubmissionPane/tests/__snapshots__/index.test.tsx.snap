// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubmissionPane renders SubmissionPane 1`] = `
<div>
  <div>
    SubmissionPane
  </div>
</div>
`;
